src/Compilers/Core/Portable/Diagnostic/ProgrammaticSuppressionInfo.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    /// <summary>
+    /// Contains information about the source of a programmatic diagnostic suppression produced by an <see cref="DiagnosticSuppressor"/>.
+    /// </summary>
+    internal sealed class ProgrammaticSuppressionInfo : IEquatable<ProgrammaticSuppressionInfo>
+    {
+        public ImmutableSortedSet<(string Id, LocalizableString Justification)> Suppressions { get; }
+
+        internal ProgrammaticSuppressionInfo(ImmutableSortedSet<(string Id, LocalizableString Justification)> suppressions)
+        {
+            // Assert that we got a sorted list of suppressions.
```

Comment:
>// Assert that we got a sorted list of suppressions. [](start = 12, length = 52)

How do we expect the list to be sorted? What is the key?

Reply:
Ah, this seems like a redundant assert from the prior implementation that passed in an `ImmutableArray` instead of `ImmutableSortedSet`. The set is sorted by the the default comparer for tuples `(string Id, LocalizableString Justification)`